src/world.cpp

```diff
@@ -2208,7 +2208,7 @@ void World_GenRoom(struct room_s *room, class VT_Level *tr)
     room->alternate_room = NULL;
     room->base_room = NULL;
 
-    if((tr_room->alternate_room >= 0) && ((uint32_t)tr_room->alternate_room < tr->rooms_count))
+    if((tr_room->alternate_room >= 0) && ((uint32_t)tr_room->alternate_room < tr->rooms_count) && (room->id < tr_room->alternate_room))
```

Comment:
Yeah, maybe some example with the alternative ids [0..4].
[0, 1, 2, 3, 4] -> swap ->[1, -1, 3, -1, -1]

Basically:
```
foreach(room1 in rooms) { // in ascending order
  if(room1->alternative < 0) continue;
  room2 = &rooms[room1->alternative];
  // move items/blocks/etc. from room1 to room2, discarding every dynamic item in room2
  swap(*room1, *room2); // swap everything except the 'alternative' flag
  swap(room1->alternative, room2->alternative);
  room2->alternative = -1;
}
```